src/scripts/filtering.js

```diff
@@ -36,18 +58,57 @@ const filterPosts = async posts => {
   }
 };
 
+let putLabelInParent
+
 export const main = async () => {
-  ({ ownPosts, duplicatePosts, filterText, filterTags, filterCws } = await getOptions('filtering'));
+  ({ ownPosts, hideShares, duplicatePosts, filterText, filterTags, filterCws } = await getOptions('filtering'));
   [filterText, filterTags, filterCws] = [filterText, filterTags, filterCws].map(list => list.trim() === '' ? false : list.split(',').map(str => str.trim().toLowerCase()));
   if (!ownPosts && duplicatePosts === 'disabled' && !filterText && !filterTags && !filterCws) return;
```

Comment:
`!hideShares` should be added to this list of checks, otherwise it won't run the feature if only hide shares is enabled